packages/cdktf-cli/lib/client/react.tsx

```diff
@@ -0,0 +1,97 @@
+import fetch from "cross-fetch";
+import { default as WebSocket } from "ws";
+import {
+  ApolloClient,
+  HttpLink,
+  InMemoryCache,
+  ApolloProvider,
+  split,
+  from,
+  NormalizedCacheObject,
+} from "@apollo/client";
+import { getMainDefinition } from "@apollo/client/utilities";
+import { WebSocketLink } from "@apollo/client/link/ws";
+import { onError } from "@apollo/client/link/error";
+import { logger } from "../logging";
+import { Text } from "ink";
+import { startServer } from "../server";
+import React, { ReactNode, useEffect, useState } from "react";
+import { SubscriptionClient } from "subscriptions-transport-ws";
+
+const useAsyncEffect = (fun: () => Promise<void>, deps: Array<any>) => {
```

Comment:
Why is this here, I don't understand the need for this function :)

Reply:
Only to save myself a level of indentation because I cannot directly pass an async function to useEffect 😅

Reply:
ah ok